### DISCUSSION TITLE
Thank you for making the first understandable parser combinator

### DISCUSSION THREAD

**Discussion by dnaka91** [winnow-rs/winnow#292] [Q&A]
Just wanted to give a big thank you for really making the first parser combinator that I can understand :tada:.

I'm pretty new to parser combinators, and several times have I tried to grok the `nom` and `chumsky` crate. Both of them have some nice tutorials, but I always got stuck at some point, as they seem to be written from a standpoint that you already know about parser combinators to some extent. Therefore, I always got stuck for anything that goes beyond what's shown in the examples.

Just read through the tutorial, which is really beginner-friendly, and it's the first time I could understand it all the way through from start to end. Especially the focus on explaining why things are done a certain way, rather than just the how.

---

In extension of that, I always had an idea to make custom data format with external schema (like Protobuf), but was scared to actually start it off because I have to parse the schema at some point, and then I'd have to understand parser combinators to make a decent parser.

Thanks to the great documentation, tutorial and overall really approachable API, I finally feel confident to write a parser for my data format schema.

**Comment by epage** [comment]
Glad to hear this is working out, not just being that its helping you but that you are doing things you wouldn't have been able to otherwise!

**Comment by epage** [comment]
As for the tutorial, a lot of that is thanks to https://tfpk.github.io/nominomicon/ which was the starting point.

**Reply by tfpk** [reply]
Glad I could provide some help :) Great work on this project @epage, and thanks for letting me know about this!
